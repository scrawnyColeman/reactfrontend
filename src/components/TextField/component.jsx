import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { default as MaterialTextField } from '@material-ui/core/TextField';
import { materialStyles, textFieldOverrideStyles } from '../../constants/styles';

const TextField = ({ defaultValue, onChange, disabled = false, multiline = false, rows = 1, placeholder = '' }) => {
    const useStyles = makeStyles(materialStyles);
    const classes = useStyles();

    return (
        <MaterialTextField
            className={classes.textfield}
            defaultValue={defaultValue}
            id="standard-multiline-flexible"
            multiline={multiline}
            rowsMax={rows}
            rows={rows}
            onChange={onChange}
            placeholder={placeholder}
            style={textFieldOverrideStyles}
            disabled={disabled}
        />
    );
};

export default TextField;
