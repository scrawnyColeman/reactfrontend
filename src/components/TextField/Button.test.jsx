import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import TextField from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('TextField component', () => {
    const onChangeMock = jest.fn();

    beforeEach(() => {
        onChangeMock.mockReset();
    });

    test('primary TextField matches snapshot', () => {
        const wrapper = shallow(<TextField placeholder="primary TextField" />);
        expect(wrapper).toMatchSnapshot();
    });
    test('secondary TextField matches snapshot', () => {
        const wrapper = shallow(<TextField hierarchy="secondary" placeholder="Second TextField" />);
        expect(wrapper).toMatchSnapshot();
    });
    test('large TextField matches snapshot', () => {
        const wrapper = shallow(<TextField size="large" placeholder="Second TextField" />);
        expect(wrapper).toMatchSnapshot();
    });
    test('multiline TextField matches snapshot', () => {
        const wrapper = shallow(<TextField size="large" placeholder="Second TextField" multiline={true} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('onChange prop is called when TextField is clicked', () => {
        const wrapper = shallow(<TextField hierarchy="secondary" placeholder="TextField" onChange={onChangeMock} />);
        wrapper.props().onChange();
        expect(onChangeMock).toHaveBeenCalledTimes(1);
    });
});
