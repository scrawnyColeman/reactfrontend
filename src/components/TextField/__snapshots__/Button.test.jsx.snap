// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component large TextField matches snapshot 1`] = `
<WithStyles(ForwardRef(TextField))
  id="standard-multiline-flexible"
  multiline={false}
  placeholder="Second TextField"
  rows={1}
  rowsMax={1}
  style={
    Object {
      "backgroundColor": "white",
      "border": "4px solid #B4DF4E",
      "boxSizing": "border-box",
      "color": "#eee",
      "margin": "0 auto",
      "padding": "8px 12px 16px",
      "textTransform": "unset",
      "width": "100%",
    }
  }
/>
`;

exports[`TextField component multiline TextField matches snapshot 1`] = `
<WithStyles(ForwardRef(TextField))
  id="standard-multiline-flexible"
  multiline={true}
  placeholder="Second TextField"
  rows={1}
  rowsMax={1}
  style={
    Object {
      "backgroundColor": "white",
      "border": "4px solid #B4DF4E",
      "boxSizing": "border-box",
      "color": "#eee",
      "margin": "0 auto",
      "padding": "8px 12px 16px",
      "textTransform": "unset",
      "width": "100%",
    }
  }
/>
`;

exports[`TextField component primary TextField matches snapshot 1`] = `
<WithStyles(ForwardRef(TextField))
  id="standard-multiline-flexible"
  multiline={false}
  placeholder="primary TextField"
  rows={1}
  rowsMax={1}
  style={
    Object {
      "backgroundColor": "white",
      "border": "4px solid #B4DF4E",
      "boxSizing": "border-box",
      "color": "#eee",
      "margin": "0 auto",
      "padding": "8px 12px 16px",
      "textTransform": "unset",
      "width": "100%",
    }
  }
/>
`;

exports[`TextField component secondary TextField matches snapshot 1`] = `
<WithStyles(ForwardRef(TextField))
  id="standard-multiline-flexible"
  multiline={false}
  placeholder="Second TextField"
  rows={1}
  rowsMax={1}
  style={
    Object {
      "backgroundColor": "white",
      "border": "4px solid #B4DF4E",
      "boxSizing": "border-box",
      "color": "#eee",
      "margin": "0 auto",
      "padding": "8px 12px 16px",
      "textTransform": "unset",
      "width": "100%",
    }
  }
/>
`;
